crates/sargon/src/system/sargon_os/pre_authorization/sargon_os_pre_authorization_status.rs

```diff
@@ -0,0 +1,360 @@
+use crate::prelude::*;
+use std::time::Duration;
+
+// ==================
+// Poll PreAuthorization Status (Public)
+// ==================
+impl SargonOS {
+    /// Polls the status of a `SubintentHash` until it is either `Success` or `Expired`.
+    pub async fn poll_pre_authorization_status(
+        &self,
+        intent_hash: SubintentHash,
+        expiration: DappToWalletInteractionSubintentExpiration,
+    ) -> Result<PreAuthorizationStatus> {
+        let (status, _) = self
+            .poll_pre_authorization_status_with_delays(intent_hash, expiration)
+            .await?;
+
+        Ok(status)
```

Comment:
Absolutely nothing wrong with the code:

```rust
let (status, _) = self
	.poll_pre_authorization_status_with_delays(intent_hash, expiration)
	.await?;

Ok(status)
```
that you have here.

But I prefer:

```rust
self
	.poll_pre_authorization_status_with_delays(intent_hash, expiration)
	.await
	.map(|status, _| status)
```

A bit shorter :) a more "functional" which we try to make use of in Sargon :)